packages/typography/src/to-theme.ts

```diff
@@ -0,0 +1,256 @@
+// custom implementation of typography.js for use in theme-ui
+import verticalRhythm from 'compass-vertical-rhythm'
+import ms from 'modularscale'
+import styles from './styles'
+
+import { TypographyOptions } from 'typography'
+import { Merge } from 'type-fest'
+
+// Temporary hack waiting for DT PR
+// https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44513
+// TODO: Replace hack with official definition
+type VerticalRhythm = ReturnType<typeof verticalRhythm>
+type VerticalRhythmOptions = Parameters<typeof verticalRhythm>[0]
+
+const unwantedTypographyOptions = [
+  'headerColor',
+  'bodyColor',
+  'overrideStyles',
+  'overrideThemeStyles',
+  'plugins',
+] as const
+type UnwantedTypographyOptions = typeof unwantedTypographyOptions[number]
+
+type BaseTypographyOptions = Omit<TypographyOptions, UnwantedTypographyOptions>
+
+interface ChangedTypographyOptions {
+  baseFontSize: number
+}
+
+export interface CustomTypographyOptions extends Merge<
+  Required<BaseTypographyOptions>,
+  ChangedTypographyOptions
+> { }
+
+export interface CustomVerticalRhythm extends VerticalRhythm {
+  options: CustomTypographyOptions
+}
+
+// - uses unitless values
+// - creates base theme object
+// - uses a static theme.styles object for consumption in theme-ui
+// - ignores overrideThemeStyles
+// - does not include color styles
+// - should be mostly compatible with existing typography.js themes
+
+const defaults: CustomTypographyOptions = {
+  baseFontSize: 16,
+  baseLineHeight: 1.45,
+  headerLineHeight: 1.1,
+  scaleRatio: 2,
+  googleFonts: [],
+  headerFontFamily: [
+    '-apple-system',
+    'BlinkMacSystemFont',
+    'Segoe UI',
+    'Roboto',
+    'Oxygen',
+    'Ubuntu',
+    'Cantarell',
+    'Fira Sans',
+    'Droid Sans',
+    'Helvetica Neue',
+    'sans-serif',
+  ],
+  bodyFontFamily: ['georgia', 'serif'],
+  headerWeight: 'bold',
+  bodyWeight: 'normal',
+  boldWeight: 'bold',
+  includeNormalize: true,
+  blockMarginBottom: 1,
+}
+
+// TODO: find better theme definition
```

Comment:
Aren't you looking for this? https://github.com/system-ui/theme-ui/blob/9388b459d8284c0a02ca99435106adc7586a6b92/packages/css/src/types.ts#L513

@theme-ui/typography is sort of a plugin, so it could probably augment the Theme.
```ts
declare module '@theme-ui/css' {
  export interface Theme {
    typography: Typography
  }
}
```

Reply:
Exactly for `Theme` interface. Giving this a try in the week.

Thanks bud :)